dirty_cat/datasets/fetching.py

```diff
@@ -128,9 +127,13 @@ def _download_and_write_openml_dataset(dataset_id: int, data_directory: str) ->
     ----------
     dataset_id: int
         The ID of the dataset to download.
+    data_directory: str
+        The directory in which the data will be saved.
 
     """
-    # The ``fetch_openml()`` function returns a scikit-learn Bunch object,
+    from sklearn.datasets import fetch_openml
```

Comment:
Is there a reason why you moved the import locally?

Typically, local imports are frowned upon. If there is a good reason, indicate it here with a comment.

Reply:
You're absolutely right, however, importing the function ``fetch_openml`` locally is the only way I found to be able to mock it in the test file.

When importing the function in the file header (like I did in the first commit), the test ``test__download_and_write_openml_dataset()`` raises an AssertionError at ``mock_fetch_openml.assert_called_once_with(data_id=1, data_home=test_data_dir)`` (line 149).

I moved the import back in the header and launching the test gives this screen :

![Screenshot from 2020-10-29 10-38-37](https://user-images.githubusercontent.com/33222654/97551241-048bd980-19d3-11eb-91e2-c880507856b9.png)

Perhaps there is another mocking method fitting the case.

PS: the comment ``# The following returns "Called 0 times" although I'm pretty sure it does get called...`` line 148 is outdated.